client/app/scripts/superdesk/editor/editor.js

```diff
@@ -550,16 +550,375 @@ function EditorService(spellcheck, $rootScope, $timeout) {
+                            vm.setFocusOnBlock(vm.blocks[block_position - 1]);
+                        } else {
+                            // if it's the first block, just remove the content
+                            block.body = '';
+                        }
+                        vm.commitChanges();
+                    },
+                    setFocusOnBlock: function(block) {
+                        vm.blocks.forEach(function(b) {
+                            b.focus = b === block;
+                        });
+                    },
+                    focusPreviousBlock: function(block) {
+                        var pos = vm.getBlockPosition(block);
+                        // if not the first one, focus on the previous
+                        if (pos > 0) {
+                            vm.setFocusOnBlock(vm.blocks[pos - 1]);
+                        }
+                    },
+                    focusNextBlock: function(block) {
+                        var pos = vm.getBlockPosition(block);
+                        // if not the last one, focus on the next
+                        if (pos + 1 < vm.blocks.length) {
+                            vm.setFocusOnBlock(vm.blocks[pos + 1]);
+                        }
+                    },
+                    getPreviousBlock: function(block) {
+                        var pos = vm.getBlockPosition(block);
+                        // if not the first one
+                        if (pos > 0) {
+                            return vm.blocks[pos - 1];
+                        }
+                    }
+                });
+            },
+            link: function(scope, element, attr, controllers) {
+                var controller = controllers[0];
+                var ngModel = controllers[1];
+                $timeout(function() {
+                    if (scope.config.multiBlockEdition) {
+                        controller.initEditorWithMultipleBlock(ngModel);
+                    } else {
+                        controller.initEditorWithOneBlock(ngModel);
+                    }
+                });
+            }
+        };
+    }])
+    .directive('sdTextEditorBlockEmbed', ['$timeout', function ($timeout) {
+        return {
+            scope: {type: '=', config: '=', language: '=', sdTextEditorBlockEmbed: '='},
+            require: ['sdTextEditorBlockEmbed', 'ngModel'],
+            template: [
+                '<textarea ng-show="vm.editable" ng-model="vm.embedCode" class="embed-editor"></textarea>',
+                '<div ng-show="!vm.editable" class="embed-preview"></div>',
+                '<a class="btn btn-default btn-small" ng-show="!vm.editable" ng-click="vm.toggleEdition()">Edit the code</a>',
+                '<a class="btn btn-default btn-small" ng-show="vm.editable" ng-click="vm.cancel()">Cancel</a>',
+                '<a class="btn btn-default btn-small" ng-show="vm.editable" ng-click="vm.save()">Save</a>'].join(''),
```

Comment:
Should we have the template in its own file? IMO more readable, since it's not that short...